File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/CsvImportResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class CsvImportResponse 
{
    public bool Valid { get; set; }
    public string Message { get; set; }
    public double VarVersion { get; set; }
    public bool SerializeNulls { get; set; }
    public long StartTimeLog { get; set; }
    public string ErrorCode { get; set; }
    public List<NameStringValueResponse> Request { get; set; }
    public long BatchId { get; set; }
    
    /// <summary>
    /// Gets or Sets CsvImportBatchStatus
    /// </summary>
    public enum CsvImportBatchStatusEnum
    {
        
        /// <summary>
        /// Enum NEWEnum for NEW
        /// </summary>
        NEWEnum = 1,
        
        /// <summary>
        /// Enum ERROREnum for ERROR
        /// </summary>
        ERROREnum = 2,
        
        /// <summary>
        /// Enum COMPLETEEnum for COMPLETE
        /// </summary>
        COMPLETEEnum = 3,
        
        /// <summary>
        /// Enum PARSEERROREnum for PARSE_ERROR
        /// </summary>
        PARSEERROREnum = 4,
        
        /// <summary>
        /// Enum PROCESSINGEnum for PROCESSING
        /// </summary>
        PROCESSINGEnum = 5,
        
        /// <summary>
        /// Enum DUPLICATEEnum for DUPLICATE
        /// </summary>
        DUPLICATEEnum = 6
    }

    public CsvImportBatchStatusEnum CsvImportBatchStatus { get; set; }
    public string ErrorMessage { get; set; }
    
    /// <summary>
    /// Gets or Sets ObjectType
    /// </summary>
    public enum ObjectTypeEnum
    {
        
        /// <summary>
        /// Enum OFFERSEnum for OFFERS
        /// </summary>
        OFFERSEnum = 1,
        
        /// <summary>
        /// Enum RETAILERSEnum for RETAILERS
        /// </summary>
        RETAILERSEnum = 2,
        
        /// <summary>
        /// Enum RETAILERLOCATIONSEnum for RETAILERLOCATIONS
        /// </summary>
        RETAILERLOCATIONSEnum = 3,
        
        /// <summary>
        /// Enum CATEGORIESEnum for CATEGORIES
        /// </summary>
        CATEGORIESEnum = 4,
        
        /// <summary>
        /// Enum FILTERSEnum for FILTERS
        /// </summary>
        FILTERSEnum = 5
    }

    public ObjectTypeEnum ObjectType { get; set; }
    public string Returning { get; set; }
}

File: server-scala-play-server/app/model/CsvImportResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for CsvImportResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class CsvImportResponse(
  valid: Option[Boolean],
  message: Option[String],
  version: Option[Double],
  serializeNulls: Option[Boolean],
  startTimeLog: Option[Long],
  errorCode: Option[String],
  request: Option[List[NameStringValueResponse]],
  batchId: Option[Long],
  csvImportBatchStatus: Option[CsvImportResponse.CsvImportBatchStatus.Value],
  errorMessage: Option[String],
  objectType: Option[CsvImportResponse.ObjectType.Value],
  returning: Option[String]
)

object CsvImportResponse {
  implicit lazy val csvImportResponseJsonFormat: Format[CsvImportResponse] = Json.format[CsvImportResponse]

  // noinspection TypeAnnotation
  object CsvImportBatchStatus extends Enumeration {
    val NEW = Value("NEW")
    val ERROR = Value("ERROR")
    val COMPLETE = Value("COMPLETE")
    val PARSEERROR = Value("PARSE_ERROR")
    val PROCESSING = Value("PROCESSING")
    val DUPLICATE = Value("DUPLICATE")

    type CsvImportBatchStatus = Value
    implicit lazy val CsvImportBatchStatusJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }

  // noinspection TypeAnnotation
  object ObjectType extends Enumeration {
    val OFFERS = Value("OFFERS")
    val RETAILERS = Value("RETAILERS")
    val RETAILERLOCATIONS = Value("RETAILERLOCATIONS")
    val CATEGORIES = Value("CATEGORIES")
    val FILTERS = Value("FILTERS")

    type ObjectType = Value
    implicit lazy val ObjectTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

